core/src/project/from_arc.rs

```diff
@@ -1,6 +1,6 @@
 use super::*;
 
-use crate::alloc::Arc;
+use std::sync::Arc;
```

Comment:
Dumb question: in many places you're switching to `core` from `std`.  Given that you need `std` here, is there any point to switching to `core` elsewhere?  Or is this chunk optional?

Reply:
Before I had `extern core as std`, so those were really using core. Which wasn't the smartest way to do things. Anything that required `std` or `alloc` was gated behind `cfg(not(feature = "no_std"))` which was awkward. I swtiched feature gates to `std`, and `alloc` instead of `no_std` and `alloc`. This now keeps in line with Rust's API guidelines.

Now the api works like so:
* When you have no features enabled (`--no-default-features`), then the `std` identifier is inaccessible (guarded by `cfg(feature ="alloc")` or `cfg(feature = "std")`).
* If you have `alloc` enabled, but not `std` (`--no-default-features --features alloc`) then at the crate root there is `extern crate alloc as std;`, which aliases `alloc` as std (so we still don't have access to os-specific operations, like `std::fs`). But we do have access to `alloc::{alloc, dealloc}`, things like `Box`, and `Vec`. One more important difference between `alloc` and `std` is the lack of prelude in `alloc`. This means you have to import `Box` and `Vec` if you want to use them, as seen [here](https://github.com/RustyYato/generic-field-projection/blob/d1538bd4453a5172900f50263fdcbe39290c70f5/core/src/project/from_box.rs#L9).
* If you have `std` enabled, you also enable `alloc`, so all the guards that just have `cfg(feature = "alloc")` are enabled, and if we need to add anything `std`-specific we can. Currently, there is nothing guarded by `std` alone.

The aliasing allows me to ignore the distinction between `alloc` and `std` if I only care about things in alloc, and just use the `std::` prefix for both. Which is nice, because that distinction is really annoying to work around in other ways. (Previously I use two separate modules that re-exported common types from `alloc` and `std`, but this is now unnecessary). I've done this in some other crates I've prototyped/built and it works out quite nicely.

Reply:
Neat!  I always assumed that you meant for this to work in std only, but I like that you're making it available in other environments.